craft_application/commands/lifecycle.py

```diff
@@ -410,10 +411,26 @@ def _run(
             package_names = ", ".join(pkg.name for pkg in packages)
             emit.progress(f"Packed: {package_names}", permanent=True)
 
+        self._services.lifecycle.save_pack_state([p.name for p in packages])
+
         if shell_after:
             _launch_shell()
 
 
+class TestCommand(PackCommand):
```

Comment:
Saying that the Test command is a Pack command feels like a stretch, doesn't it? From what I can tell, this command class here doesn't do anything by itself - it signals to the inner instance that it needs to "pack", and to the outer instance that it needs to test (in the Application, not in the command). This feels very confusing

Reply:
This is also a shortcut because to do it properly we need to handle state here and only run prime if needed.

Reply:
Making this a lifecycle command is really a limitation of the outer command being tied to the managed command, right? For the `make test` analogy, what we need here is the output of pack.